### ISSUE TITLE
Rulers position is incorrect when using window.zoomLevel

### ISSUE THREAD

**Issue by jsnjack** [Microsoft/vscode#18356]
- VSCode Version: 1.8.1
- OS Version: Fedora 25

Steps to Reproduce:

1.  Set window.zoomLevel = 0.2 and editor.rulers = [80]
2. The actual position of the ruler is 84 instead of 80

**Comment by alexandrudima** [comment]
I cannot reproduce. @jsnjack can you please give us more information to be able to reproduce?

![image](https://cloud.githubusercontent.com/assets/5047891/21850169/8d69498c-d809-11e6-8eb7-4b9511660fc2.png)

**Comment by jsnjack** [comment]
That is my configuration:
![screenshot from 2017-01-11 14-30-04](https://cloud.githubusercontent.com/assets/5493713/21850465/fb059aa8-d80a-11e6-952d-7c8e75b77c0c.png)

**Comment by alexandrudima** [comment]
Is the font monospace? (I don't have Fedora installed at this time)

Is it possible to ensure you have a monospace font? e.g. Inconsolata

**Comment by jsnjack** [comment]
I use Dejavu Sans Mono font. But you are right, it is related to the fonts. My user settings are:
```
    "editor.fontFamily": "DejaVu Sans Mono",
    "window.zoomLevel": 0.2,
    "editor.fontSize": 13,
    "editor.rulers": [80, 120],
```
If I have empty workspace config, the ruler is located in column 84. If I specify `"editor.fontFamily": "DejaVu Sans Mono"` in the workspace config, the ruler is in the correct position

**Comment by jsnjack** [comment]
Correction: If I specify "editor.fontFamily": "DejaVu Sans Mono" in the workspace config, the ruler is in the column 81

**Comment by alexandrudima** [comment]
@jsnjack I personally recommend staying away from `window.zoomLevel` if you would like to get pixel perfect or antialiased rendering and simply use `editor.fontSize`.

`window.zoomLevel` is the equivalent of zooming in your browser. It eventually causes all our nice integer rendering coordinates to be multiplied by a floating point, leading to all sorts of rounding errors and blurriness.

**TL;DR** If you want pixel perfect rendering, do not use `window.zoomLevel`.

See https://github.com/Microsoft/vscode/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aopen%20zoomLevel

**Comment by jsnjack** [comment]
@alexandrudima I do indeed have a lot of problems with zoomLevel :( However, I have another problem which is explained here #1896 (blurry fonts on 4k monitor). Setting zoomLevel helps to have sharp fonts. Is there a better way to get sharp fonts?

**Comment by alexandrudima** [comment]
The only behavioural difference on our side that I can think of when having `window.zoomLevel !== 0` is that we don't avoid `translate3d` (exactly to avoid bluriness). Would the setting `"editor.disableTranslate3d": true` make a difference for you?

**Comment by jsnjack** [comment]
`"editor.disableTranslate3d": true ` makes no difference for me. the font is still blurry
